backend/dep/provider/search.go

```diff
@@ -0,0 +1,11 @@
+package provider
+
+import "time"
+
+// SearchAPITimeout represents timeout duration of a search request.
+type SearchAPITimeout time.Duration
+
+// NewSearchAPITimeout creates Duration.
+func NewSearchAPITimeout(timeout SearchAPITimeout) time.Duration {
+	return time.Duration(timeout)
+}
```

Comment:
```suggestion
```

Reply:
The wire generation fails without this because it can't find provider for time.Duration

Reply:
It should be:
```suggestion
func NewSearch(
        logger logger.Logger,
        shortLinkRepo repository.ShortLink,
	userShortLinkRepo repository.UserShortLink,
	timeout SearchTimeout,
	) search.Search {
	// Please also reorder the parameter of the constructor of search
	return NewSearch(logger, shortLinkRepo, userShortLinkRepo, timeout)
}
```

Reply:
Thank you for your help!